src/libasr/codegen/asr_to_c_cpp.h

```diff
@@ -198,6 +199,27 @@ R"(#include <stdio.h>
         src = unit_src;
     }
 
+    void get_expr_src(ASR::expr_t *arg) {
+        self().visit_expr(*arg);
+        if (ASR::is_a<ASR::ListConstant_t>(*arg) ||
+                ASR::is_a<ASR::TupleConstant_t>(*arg) ||
+                ASR::is_a<ASR::DictConstant_t>(*arg)) {
+            src = const_var_names[get_hash((ASR::asr_t*)arg)];
+        } else if (ASR::is_a<ASR::ListSection_t>(*arg)) {
+            ASR::ListSection_t *l_sec = ASR::down_cast<ASR::ListSection_t>(arg);
+            src = "*" + const_var_names[get_hash((ASR::asr_t*)l_sec)];
+        }
```

Comment:
I think these can go into visit_ListContant, visit_TupleConstant, visit_ListSection, etc. and at the end of the function simply put the line `src = const_var_names[get_hash((ASR::asr_t*)arg)];` etc. Then remove this `get_expr_src`.